src/query/interpreter.cpp

```diff
@@ -1177,11 +1187,12 @@ Interpreter::Interpreter(InterpreterContext *interpreter_context) : interpreter_
   MG_ASSERT(interpreter_context_, "Interpreter context must not be NULL");
 }
 
-PreparedQuery Interpreter::PrepareTransactionQuery(std::string_view query_upper) {
+PreparedQuery Interpreter::PrepareTransactionQuery(std::string_view query_upper,
+                                                   const std::map<std::string, storage::PropertyValue> &metadata) {
   std::function<void()> handler;
 
   if (query_upper == "BEGIN") {
-    handler = [this] {
+    handler = [this, metadata] {
```

Comment:
Did you consider capturing `metadata` by `&`?

Reply:
The handler gets returned and called in the Pull. Unfortunately I think we have to copy so it doesn't segfault.

Reply:
This would break the current state of the code, but if you do a generalized lambda capture `[this, metadata = std::move(metadata)]` then you can move the `std::map<>` instead of a copy. For that you would need to remove the `const`-ness of the `&` from the call-chain leading up to this part.
Also, -this is not the case here but if we want to do that for some reason- if at any point you would move `metadata` out of the closure, you would need to make the lambda `mutable` as well.
Not sure how big this map can become, if copying it is not a problem it does not really matter.

Reply:
That's fine, maybe leave a TODO comment somewhere around here.